File: backend/csharp/KisiIslemleri.cs
using System;
using System.IO;
using System.Text.RegularExpressions;
using Npgsql;

public class KisiIslemleri
{
    private static string connectionString = "Host=db;Username=kullanici;Password=sifre;Database=rehber";

    public static bool ValidateEmail(string email)
    {
        string pattern = @"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$";
        Regex regex = new Regex(pattern);
        return regex.IsMatch(email);
    }

    public static void KisiEkle(string isim, string eposta, string telefon, string adres)
    {
        using (var connection = new NpgsqlConnection(connectionString))
        {
            connection.Open();
            var command = new NpgsqlCommand(
                "INSERT INTO kisi (isim, eposta, telefon, adres) VALUES (@isim, @eposta, @telefon, @adres)",
                connection
            );
            command.Parameters.AddWithValue("@isim", isim);
            command.Parameters.AddWithValue("@eposta", eposta);
            command.Parameters.AddWithValue("@telefon", telefon);
            command.Parameters.AddWithValue("@adres", adres);
            command.ExecuteNonQuery();
            Console.WriteLine("Kişi eklendi.");
        }
    }

    public static void ExportEmailsToFile(string filename = "emails_csharp.txt")
    {
        using (var connection = new NpgsqlConnection(connectionString))
        {
            connection.Open();
            var command = new NpgsqlCommand("SELECT eposta FROM kisi", connection);
            using (var reader = command.ExecuteReader())
            {
                using (StreamWriter file = new StreamWriter(filename))
                {
                    while (reader.Read())
                    {
                        string email = reader.GetString(0);
                        if (ValidateEmail(email))
                        {
                            file.WriteLine(email);
                        }
                    }
                }
            }
            Console.WriteLine($"E-posta adresleri {filename} dosyasına aktarıldı.");
        }
    }

    public static void Main(string[] args)
    {
        KisiEkle("Ahmet Yılmaz", "ahmet@example.com", "5551234567", "Ankara, Türkiye");
        ExportEmailsToFile();
    }
}

File: backend/legacy/cpp/kisi_islemleri.cpp
#include <iostream>
#include <pqxx/pqxx>
#include <fstream>
#include <regex>

using namespace std;
using namespace pqxx;

bool validateEmail(const string& email) {
    const regex pattern(R"(^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$)");
    return regex_match(email, pattern);
}

void exportEmailsToFile(const string& filename = "emails_cpp.txt") {
    try {
        connection conn("host=db user=kullanici password=sifre dbname=rehber");
        work txn(conn);
        result res = txn.exec("SELECT eposta FROM kisi");

        ofstream file(filename);
        for (auto row : res) {
            string email = row[0].as<string>();
            if (validateEmail(email)) {
                file << email << endl;
            }
        }
        cout << "E-posta adresleri " << filename << " dosyasına aktarıldı." << endl;
    } catch (const exception &e) {
        cerr << "Hata: " << e.what() << endl;
    }
}

int main() {
    try {
        connection conn("host=db user=kullanici password=sifre dbname=rehber");

        work txn(conn);
        txn.exec(
            "INSERT INTO kisi (isim, eposta, telefon, adres) "
            "VALUES ('Mehmet Demir', 'mehmet@example.com', '5559876543', 'İstanbul, Türkiye')"
        );
        txn.commit();
        cout << "Kişi eklendi." << endl;

        exportEmailsToFile();
    } catch (const exception &e) {
        cerr << "Hata: " << e.what() << endl;
        return 1;
    }
    return 0;
}
